binary-decode.go

```diff
@@ -695,18 +696,158 @@ func (cdc *Codec) decodeReflectBinaryStruct(bz []byte, info *TypeInfo, rv reflec
 
 		// Read "StructTerm".
 		// NOTE: In the future, we'll need to break out of a loop
-		// when encoutering an StructTerm typ3 byte.
-		var typ = Typ3(0x00)
+		// when encountering an StructTerm typ3 byte.
+
+		var typ Typ3
+		// we have read all, or do we? consume the rest
+		for len(bz) > 2 && bz[0]&0xF8 != 0 && Typ3(bz[0]) != Typ3_StructTerm {
```

Comment:
I think we just want a for loop with no condition here, and below in `decodeFieldNumberAndTyp3(bz)` break (without sliding) if typ3 is Typ3_StructTerm.  

Then, line 716~723 below is unneeded, as the unconditional for-loop will return an error if after sliding bz is insufficient in length.  Ergo no need for len(bz) > 2 in the if-condition.

Reply:
I added a small test to see if this works.